src/openforms/forms/tests/test_migrations.py

```diff
@@ -435,3 +435,57 @@ def test_fix_badly_formatted_rules(self):
             self.rule_badly_formatted.json_logic_trigger,
             {"==": [{"var": "selectBoxes.a"}, True]},
         )
+
+
+class TestLogicParsing(TestMigrations):
+    migrate_from = "0002_auto_20210917_1114_squashed_0045_remove_formstep_optional"
+    migrate_to = "0046_convert_advanced_logic"
+    app = "forms"
+
+    def setUpBeforeMigration(self, apps):
+        self.FormDefinition = apps.get_model("forms", "FormDefinition")
+        self.components = [
+            {
+                "type": "fieldset",
+                "key": "fieldset-1",
+                "logic": [
+                    {
+                        "trigger": {
+                            "type": "simple",
+                            "simple": {
+                                "show": True,
+                                "when": "non-existing-object",
+                                "eq": "trigger value",
+                            },
+                        },
+                        "actions": [
+                            {
+                                "type": "property",
+                                "property": {
+                                    "label": "Hidden",
+                                    "value": "hidden",
+                                    "type": "boolean",
+                                },
+                                "state": False,
+                            }
+                        ],
+                    },
+                ],
+            },
+        ]
+        self.FormDefinition.objects.create(
+            name="Definition with textfield",
+            slug="definition-with-textfield",
+            configuration={"components": self.components},
+        )
+
+    def test_parse_trigger(self):
+        """Check that :func:`parse_trigger` does not error out on retrieval of form(definition)
```

Comment:
this is some confusing phrasing - the function `parse_trigger` will never error out on retrieval of a form definition, since it's not involved in that process at all.

Rather use something like `Assert that the form definition was migrated correctly`

The section below provides sufficient context.